drivers/sensor/lis2dw12/lis2dw12.c

```diff
@@ -157,11 +158,81 @@ static int lis2dw12_config(const struct device *dev, enum sensor_channel chan,
 	return -ENOTSUP;
 }
 
+
+static inline int32_t sensor_ms2_to_mg(const struct sensor_value *ms2)
+{
+	int64_t nano_ms2 = (ms2->val1 * 1000000LL + ms2->val2) * 1000LL;
+
+	if (nano_ms2 > 0) {
+		return (nano_ms2 + SENSOR_G / 2) / SENSOR_G;
+	} else {
+		return (nano_ms2 - SENSOR_G / 2) / SENSOR_G;
+	}
+}
+
+#if CONFIG_LIS2DW12_THRESHOLD
+static int lis2dw12_attr_set_thresh(const struct device *dev,
+					enum sensor_channel chan,
+					enum sensor_attribute attr,
+					const struct sensor_value *val)
+{
+	uint8_t reg;
+	size_t ret;
+	const struct lis2dw12_device_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+
+	LOG_DBG("%s on channel %d", __func__, chan);
+
+	/* can only be set for all directions at once */
+	if (chan != SENSOR_CHAN_ACCEL_XYZ) {
+		return -EINVAL;
+	}
+
+	/* Configure wakeup threshold threshold. */
+	lis2dw12_fs_t range;
+	int err = lis2dw12_full_scale_get(ctx, &range);
+
+	if (err) {
+		return err;
+	}
+
+	uint32_t thr_mg = abs(sensor_ms2_to_mg(val));
+
+	/* Check maximum value: depends on current FS value */
+	if (thr_mg >= (2U << range) * 1000) {
+		return -EINVAL;
+	}
+
+	/* The threshold is applied to both positive and negative data:
+	 * for a wake-up interrupt generation at least one of the three axes must be
+	 * bigger than the threshold. */
+	int lsb_mg = ((2U << range) * 1000) / 64;; /* range = 2/4/8/16G */
+
+	reg = (thr_mg + (lsb_mg / 2)) / lsb_mg;
+
+	LOG_DBG("Threshold %d mg -> fs: %d g -> reg = %d LSBs", thr_mg, (2U << range), reg);
+	ret = 0;
+
+	return lis2dw12_wkup_threshold_set(ctx, reg);
+}
+#endif
+
 static int lis2dw12_attr_set(const struct device *dev,
 			      enum sensor_channel chan,
 			      enum sensor_attribute attr,
 			      const struct sensor_value *val)
 {
+#if CONFIG_LIS2DW12_THRESHOLD
+	switch (attr) {
+	case SENSOR_ATTR_UPPER_THRESH:
```

Comment:
missing `__fallthrough`

Reply:
`__fallthrough` for consecutive cases in a switch? (without any other lines of code in between)
Sure, I can do that, but:
- doesn't that hurt readability more, than help it?
- I can't see that being done almost anywhere else in `drivers/sensor`